pkg/registry/rbac/reconciliation/namespace.go

```diff
@@ -0,0 +1,42 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package reconciliation
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	utilerrors "k8s.io/apimachinery/pkg/util/errors"
+	corev1client "k8s.io/client-go/kubernetes/typed/core/v1"
+)
+
+// ensureNamespace gets or creates the given namespace while ignoring forbidden errors.
+func ensureNamespace(client corev1client.NamespaceInterface, namespace string) error {
+	_, getErr := client.Get(namespace, metav1.GetOptions{})
+	if getErr == nil {
+		return nil
+	}
+
+	if fatalGetErr := utilerrors.FilterOut(getErr, apierrors.IsNotFound, apierrors.IsForbidden); fatalGetErr != nil {
+		return fatalGetErr
+	}
+
+	ns := &corev1.Namespace{ObjectMeta: metav1.ObjectMeta{Name: namespace}}
+	_, createErr := client.Create(ns)
+
+	return utilerrors.FilterOut(createErr, apierrors.IsAlreadyExists, apierrors.IsForbidden)
```

Comment:
don't filter `IsForbidden` here, if the namespace couldn't be gotten or created, I'd expect an error to be returned to the caller. The caller can decide to fail, log, or ignore the error (in this case, I think we would either log or ignore a forbidden error?)

Reply:
alternately, if this method has the possibility to return with no error with the namespace not actually created, change the name of the method to reflect that weaker guarantee

Reply:
updated func name and godoc